### PR TITLE
BasicFileTestSuite reworked

### LINKED ISSUE #115: Refactor: Rework BasicFileTestSuite to be more in-line with newer tests
### Description

BasicFileTestSuite contains 10 facts at the moment, with multiple asserts per fact. This should be broken up into more facts to be more thorough.

### Script Name

BasicFileTestSuite 

### Steps To The Task

- [x] Split up all tests within existing facts into their own facts
- [x] Add comments
- [x] Update assert descriptions

### PR DESCRIPTION
**Pull request by AdamBorek**
Facts have been broken up into multiple smaller ones

### CODE REVIEW

**Review by Indibulous — CHANGES_REQUESTED** [review]
Test names are lacking numbers (multiple `"directory_test"`s instead of `"directory_test #1"`, `"directory_test #2"`, etc.) and assert descriptions seem to contain old numbering conventions (`"#1.1 file_exists( filepath ), file should not exist yet"` instead of `"file_exists( filepath ), file should not exist yet"` or just `"file should not exist yet"`)

Apart from this, everything seems fine! but this should probably be fixed before merging.

**Comment by AdamBorek** [comment]
> Test names are lacking numbers (multiple `"directory_test"`s instead of `"directory_test #1"`, `"directory_test #2"`, etc.) and assert descriptions seem to contain old numbering conventions (`"#1.1 file_exists( filepath ), file should not exist yet"` instead of `"file_exists( filepath ), file should not exist yet"` or just `"file should not exist yet"`)
> 
> Apart from this, everything seems fine! but this should probably be fixed before merging.

My bad, I'll fix this in a second

File: projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml [hunk 1]
```diff
@@ -329,105 +772,225 @@ function BasicFileTestSuite() : TestSuite() constructor {
...
+	addFact("file_text_open_from_string_test #7", function() {
+
+		var input, _output, _file;
+
 		//#3 file_text_open_from_string( string const ) - single 4-byte utf-8 char
 		input = "🙂";
 		_file = file_text_open_from_string(input);
 		_output = file_text_read_string(_file);
-		assert_equals(_output, "🙂", "#3 file_text_open_from_string( string:local ), failed to read single 4-byte utf-8 char")
+		assert_equals(_output, "🙂", "#ile_text_open_from_string( string:local ), failed to read single 4-byte utf-8 char")
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml]:
```suggestion
		assert_equals(_output, "🙂", "#file_text_open_from_string( string:local ), failed to read single 4-byte utf-8 char")
```

File: projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml [hunk 2]
```diff
@@ -329,105 +772,225 @@ function BasicFileTestSuite() : TestSuite() constructor {
...
+	
+	addFact("file_text_open_from_string_test #8", function() {
+
+		var input, _output, _file;
 			
 		//#4 file_text_open_from_string( string const ) - complex utf-8 string
 		_file = file_text_open_from_string("!£水🙂");
 		_output = file_text_read_string(_file);
-		assert_equals(_output, "!£水🙂", "#4 file_text_open_from_string( string:local), failed to read complex utf-8 string")
+		assert_equals(_output, "!£水🙂", "#ile_text_open_from_string( string:local), failed to read complex utf-8 string")
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml]:
```suggestion
		assert_equals(_output, "!£水🙂", "#file_text_open_from_string( string:local), failed to read complex utf-8 string")
```

File: projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml [hunk 3]
```diff
@@ -269,30 +636,64 @@ function BasicFileTestSuite() : TestSuite() constructor {
...
+		var value, directory = working_directory + "testDir";
+
+		if (platform_not_browser()) {
+
+			// Make sure test directory doesn't exist
+			if (directory_exists(directory)) directory_destroy(directory);
+			directory_create(directory);
+		}
+
+		var filename = "testFile.txt"
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml]:
```suggestion
		var filename = "testFile.txt";
```

File: projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml [hunk 4]
```diff
@@ -193,49 +384,225 @@ function BasicFileTestSuite() : TestSuite() constructor {
...
+
+		if (platform_not_browser()) {
 
-	addFact("file_text_test", function() {
+			// Make sure test directory doesn't exist
+			if (directory_exists(directory)) directory_destroy(directory);
+			directory_create(directory);
+		}
+
+		var filename = "testFile.txt"
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml]:
```suggestion
		var filename = "testFile.txt";
```

File: projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml [hunk 5]
```diff
@@ -193,49 +384,225 @@ function BasicFileTestSuite() : TestSuite() constructor {
...
+		file_bin_write_byte(_fileHandle, 1100);
+		file_bin_write_byte(_fileHandle, 0001);
+		file_bin_write_byte(_fileHandle, 1000);
+		// Close file
+		file_bin_close(_fileHandle);
+		
+		_fileHandle = file_bin_open(filePath, 0);
+			
+		//#4 file_bin_read_byte ( file )
+		_output = file_bin_read_byte(_fileHandle)
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml]:
```suggestion
		_output = file_bin_read_byte(_fileHandle);
```

File: projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml [hunk 6]
```diff
@@ -307,19 +708,61 @@ function BasicFileTestSuite() : TestSuite() constructor {
...
+		var value, directory = working_directory + "testDir";
+
+		if (platform_not_browser()) {
+
+			// Make sure test directory doesn't exist
+			if (directory_exists(directory)) directory_destroy(directory);
+			directory_create(directory);
 		}
+
+		var filename = "testFile.txt"
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml]:
```suggestion
		var filename = "testFile.txt";
```

File: projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml [hunk 7]
```diff
@@ -329,105 +772,225 @@ function BasicFileTestSuite() : TestSuite() constructor {
...
+		assert_equals(_output, @"\foo\bar\pathname\to\the", "filename_dir ( backslash_path ) failed");
+			
+	});
+
+	addFact("filename_test_simulated #9", function() {
+
+		var _output, filename = @"\foo\bar\pathname\to\the\fileTest.txt";
 			
 		_output = filename_ext(filename);
 		assert_equals(_output, ".txt", "#2.3 filename_ext ( backslash_path ) failed");
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicFileTestSuite/BasicFileTestSuite.gml]:
```suggestion
		assert_equals(_output, ".txt", "filename_ext ( backslash_path ) failed");
```

**Review by Indibulous — APPROVED** [review]
All good! Just added a few missing semicolons & fixed a couple typos. 👍

**Pull request merged**